system/lib/wasmfs/syscalls.cpp

```diff
@@ -288,6 +288,58 @@ __wasi_errno_t __wasi_fd_close(__wasi_fd_t fd) {
   return __WASI_ERRNO_SUCCESS;
 }
 
+backend_t wasmfs_get_backend_by_fd(int fd) {
+  auto fileTable = wasmFS.getLockedFileTable();
+
+  // Check that an open file exists corresponding to the given fd.
+  auto openFile = fileTable[fd];
+  if (!openFile) {
+    return NullBackend;
+  }
+
+  auto lockedOpenFile = openFile.locked();
+  return lockedOpenFile.getFile()->locked().getBackend();
+}
+
+// This function is exposed to users to allow them to obtain a backend_t for a
+// specified path.
+backend_t wasmfs_get_backend_by_path(char* path) {
+  auto pathParts = splitPath(path);
+
+  if (pathParts.empty()) {
+    return NullBackend;
+  }
+
+  // TODO: Remove this when path parsing has been re-factored.
+  if (pathParts.size() == 1 && pathParts[0] == "/") {
+    return wasmFS.getRootDirectory()->locked().getBackend();
+  }
+
+  auto base = pathParts.back();
+
+  long err;
+  auto parentDir = getDir(pathParts.begin(), pathParts.end() - 1, err);
+
+  // Parent node doesn't exist.
+  if (!parentDir) {
+    return NullBackend;
+  }
+
+  auto lockedParentDir = parentDir->locked();
+
+  // TODO: In a future PR, edit function to just return the requested file
+  // instead of having to first obtain the parent dir.
+  auto curr = lockedParentDir.getEntry(base);
+
+  if (curr) {
+    auto dir = curr->dynCast<Directory>();
+
+    return dir ? dir->locked().getBackend() : NullBackend;
```

Comment:
Casting to directory should no longer be necessary, right?

Reply:
Just rebased :)